libyul/AsmAnalysis.cpp

```diff
@@ -682,7 +654,39 @@ void AsmAnalyzer::expectValidType(string const& type, SourceLocation const& _loc
 		);
 }
 
-void AsmAnalyzer::warnOnInstructions(dev::eth::Instruction _instr, SourceLocation const& _location)
+// FIXME: copy'n'paste from AsmParser. make it more general (by putting it into dev::eth namespace)
+std::map<string, dev::eth::Instruction> const& instructions()
+{
+	// Allowed instructions, lowercase names.
+	static map<string, dev::eth::Instruction> s_instructions;
+	if (s_instructions.empty())
+	{
+		for (auto const& instruction: dev::eth::c_instructions)
+		{
+			if (
+				instruction.second == dev::eth::Instruction::JUMPDEST ||
+				dev::eth::isPushInstruction(instruction.second)
+			)
+				continue;
+			string name = instruction.first;
+			transform(name.begin(), name.end(), name.begin(), [](unsigned char _c) { return tolower(_c); });
+			s_instructions[name] = instruction.second;
+		}
+	}
+	return s_instructions;
+}
+
+bool AsmAnalyzer::warnOnInstructions(std::string const& _instructionIdentifier, langutil::SourceLocation const& _location)
+{
+	auto const& instructionMap = instructions();
```

Comment:
I think this should not use `instructions()` but rather `yul::EVMDialect::strictAssemblyForEVM(EVMVersion{}).builtin(_instructionIdentifier)`